### DISCUSSION TITLE
How do I access codespace secrets inside of docker-compose.yml?

### DISCUSSION THREAD

**Discussion by cj** [community/community#6026] [Codespaces]
Hi,

How do you access codespace secrets inside of docker-compose.yml?

I have setup a few secrets in my repo for codespaces, but I can't access the variable inside of https://github.com/microsoft/vscode-dev-containers/blob/main/container-templates/docker-compose/.devcontainer/docker-compose.yml I have tried both of these with no luck:

```json
{
  "containerEnv": {
    "HASURA_GRAPHQL_JWT_SECRET": "${localEnv:HASURA_GRAPHQL_JWT_SECRET}",
    "HASURA_GRAPHQL_ADMIN_SECRET": "${localEnv:HASURA_GRAPHQL_ADMIN_SECRET}",
    "DATABASE_URL": "${localEnv:DATABASE_URL}"
  },
  "remoteEnv": {
    "LOCAL_WORKSPACE_FOLDER": "${localWorkspaceFolder}",
    "HASURA_GRAPHQL_JWT_SECRET": "${localEnv:HASURA_GRAPHQL_JWT_SECRET}",
    "HASURA_GRAPHQL_ADMIN_SECRET": "${localEnv:HASURA_GRAPHQL_ADMIN_SECRET}",
    "DATABASE_URL": "${localEnv:DATABASE_URL}"
  },
}
```

Cheers!

**Comment by 2percentsilk** [comment]
Hi there! It looks like you're trying to reference secrets during container creation and secrets are only available after creation completes so what you're attempting to do is not currently supported.

We're currently working on an update that will set secrets as environment variables during the `initializeCommand` (https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_lifecycle-scripts). After that update is shipped, you should be able to reference your secrets as you're doing in the `containerEnv` section.
